plugin.py

```diff
@@ -1,62 +1,69 @@
 import os
 import hashlib
 import shutil
-import threading
```

Comment:
Downloading and installing in the background is handled by LSP?

Reply:
Yes, this usage pattern is so common that we might as well put it into the base package using the `needs_update_or_installation` and `install_or_update` classmethods.

Reply:
Sounds good with regards of LSP waiting for a download before trying to start the server.